### ISSUE TITLE
Weird mesh update behavior 

### ISSUE THREAD

**Issue by trsh** [FlaxEngine/FlaxEngine#498]
https://github.com/trsh/FlaxVoxelPlugin

problem in repo readme.

**Edit:**

It feels like some problem with `UpdateMesh` in editor mode

**Comment by mafiesto4** [comment]
It looks the same or very similar to me when I comment out lines 138-140 in `MC.cpp`. Could you post screenshot with an explanation?

**Comment by trsh** [comment]
With lines 

![1](https://user-images.githubusercontent.com/3727523/118262266-50105b80-b4bd-11eb-993e-641743a75246.png)

Without after some recompiles:

![2](https://user-images.githubusercontent.com/3727523/118262320-63bbc200-b4bd-11eb-9e26-c5aaa30df0f3.png)

can better see in normal view:

![3](https://user-images.githubusercontent.com/3727523/118262356-746c3800-b4bd-11eb-9a7a-99d1720214ed.png)

**Comment by trsh** [comment]
Also after X recompiles in content windows the scene just closes

![laaaa](https://user-images.githubusercontent.com/3727523/128522089-f7fef4dd-4bf1-4b5b-9f2e-777a64788df7.png)

**Comment by mafiesto4** [comment]
Thanks for reporting this!

Okay, I checked it and:
- You shouldn't normalize Normal vector after packing it into 0-1 range (line `122` in `MC.cpp`, `rn.Normalize()` is wrong)
- The scene was missing from the window because you have bug in your code that double-registers from scene vent (line `126` in `VoxelEditor.cs`, Initialize/Deinitialize handles this just fine). It's kind of valid for C# but those events are generated from C++ code. I added check to bindings generator so it will throw error exception next time: d65a85e30ceacc8887c2b4e387153ccad1573c61
- In Debug mode it works fine but in Development the bug appears -> strange!

After spending more time with it and profiling with RenderDoc I noticed that if I disable shaders optimization for debugging it works fine so I suspected some shader stuff and it appears that it was caused by having some random data set to `LightmapUVs` and `TexCoord` so when GPU was reading data it could have some random trash that was causing issued for random triangles.
Fixed by adding somewhere around vertex data initialization in your mesher:
```cpp
vb1.LightmapUVs = Half2::Zero;
vb1.TexCoord = Half2::Zero;
```

Hope it works fine now!
